contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,944 @@
+    uint160 public immutable sqrtRatioX96TickLower;
+    /// @notice sqrtRatioX96TickHigher
+    /// @dev tick higher has value 0 and represents 1:1 price parity of WETH to OETHb
+    uint160 public immutable sqrtRatioX96TickHigher;
+    /// @dev tick closest to 1:1 price parity
+    ///      Correctly assesing which tick is closer to 1:1 price parity is important since it affects
+    ///      the way we calculate the underlying assets in check Balance
+    uint160 public immutable sqrtRatioX96TickClosestToParity;
+
+    error NotEnoughWethForSwap(uint256 wethBalance, uint256 requiredWeth); // 0x989e5ca8
+    error NotEnoughWethLiquidity(uint256 wethBalance, uint256 requiredWeth); // 0xa6737d87
+    error PoolRebalanceOutOfBounds(
+        uint256 currentPoolWethShare,
+        uint256 requiredPoolWethShare
+    ); // 0x6c6108fb
+    error OutsideExpectedTickRange(int24 currentTick); // 0x46a58db6
+
+    event PoolRebalanced(
+        uint256 currentPoolWethShare,
+        uint256 targetPoolWethShare
+    );
+
+    event PoolWethShareUpdated(uint256 newWethShare);
+
+    event PrincipalPositionAfterSwap(
+        uint256 wethPositionBalance,
+        uint256 oethbPositionBalance
+    );
+
+    event PoolWethShareVarianceAllowedUpdated(
+        uint256 poolWethShareVarianceAllowed
+    );
+
+    event LiquidityRemoved(
+        uint256 withdrawLiquidityShare,
+        uint256 removedWETHAmount,
+        uint256 removedOETHbAmount,
+        uint256 wethAmountCollected,
+        uint256 oethbAmountCollected,
+        uint256 underlyingAssets
+    );
+
+    event LiquidityAdded(
+        uint256 wethAmountDesired,
+        uint256 oethbAmountDesired,
+        uint256 wethAmountSupplied,
+        uint256 oethbAmountSupplied,
+        uint256 tokenId,
+        uint256 underlyingAssets
+    );
+
+    event UnderlyingAssetsUpdated(uint256 underlyingAssets);
+
+    /**
+     * @dev Verifies that the caller is the Governor, or Strategist.
+     */
+    modifier onlyGovernorOrStrategist() {
+        require(
+            msg.sender == governor() ||
+                msg.sender == IVault(vaultAddress).strategistAddr(),
+            "Not the Governor or Strategist"
+        );
+        _;
+    }
+
+    /**
+     * @dev Un-stakes the token from the gauge for the execution duration of
+     * the function and after that re-stakes it back in.
+     *
+     * It is important that the token is unstaked and owned by the strategy contract
+     * during any liquidity altering operations and that it is re-staked back into the
+     * gauge after liquidity changes. If the token fails to re-stake back to the
+     * gauge it is not earning incentives.
+     */
+    modifier gaugeUnstakeAndRestake() {
+        if (tokenId != 0) {
+            clGauge.withdraw(tokenId);
+        }
+        _;
+        if (tokenId != 0) {
+            positionManager.approve(address(clGauge), tokenId);
+            clGauge.deposit(tokenId);
```

Comment:
🟠 Looks like it'll stake an empty liquidity position NFT on the Gauge after a `withdrawAll()`. I'm not sure if that breaks something though

Reply:
Yup it does. fix for it here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/f551d7f37aa891880d3ed0893a7d894446680967